l1-contracts-foundry/script/DeployL1.s.sol

```diff
@@ -0,0 +1,703 @@
+            genesisBatchHash: config.contracts.genesisRoot,
+            genesisIndexRepeatedStorageChanges: uint64(config.contracts.genesisRollupLeafIndex),
+            genesisBatchCommitment: config.contracts.genesisBatchCommitment,
+            diamondCut: diamondCut,
+            protocolVersion: config.contracts.latestProtocolVersion
+        });
+
+        address contractAddress = deployViaCreate2(
+            abi.encodePacked(
+                type(TransparentUpgradeableProxy).creationCode,
+                abi.encode(
+                    addresses.stateTransition.stateTransitionImplementation,
+                    addresses.transparentProxyAdmin,
+                    abi.encodeCall(StateTransitionManager.initialize, (diamondInitData))
+                )
+            )
+        );
+        console.log("StateTransitionManagerProxy deployed at:", contractAddress);
+        addresses.stateTransition.stateTransitionProxy = contractAddress;
+    }
+
+    function registerStateTransitionManager() internal {
+        Bridgehub bridgehub = Bridgehub(addresses.bridgehub.bridgehubProxy);
+        vm.broadcast();
+        bridgehub.addStateTransitionManager(addresses.stateTransition.stateTransitionProxy);
+        console.log("StateTransitionManager registered");
+    }
+
+    function setStateTransitionManagerInValidatorTimelock() internal {
+        ValidatorTimelock validatorTimelock = ValidatorTimelock(addresses.validatorTimelock);
+        vm.broadcast();
+        validatorTimelock.setStateTransitionManager(
+            IStateTransitionManager(addresses.stateTransition.stateTransitionProxy)
+        );
+        console.log("StateTransitionManager set in ValidatorTimelock");
+    }
+
+    function deployDiamondProxy() internal {
+        Diamond.FacetCut[] memory facetCuts = new Diamond.FacetCut[](1);
+        facetCuts[0] = Diamond.FacetCut({
+            facet: addresses.stateTransition.adminFacet,
+            action: Diamond.Action.Add,
+            isFreezable: false,
+            selectors: Utils.getAllSelectors(addresses.stateTransition.adminFacet.code)
+        });
+        Diamond.DiamondCutData memory diamondCut = Diamond.DiamondCutData({
+            facetCuts: facetCuts,
+            initAddress: address(0),
+            initCalldata: hex""
+        });
+        bytes memory bytecode = abi.encodePacked(
+            type(DiamondProxy).creationCode,
+            abi.encode(config.chainId, diamondCut)
+        );
+        address contractAddress = deployViaCreate2(bytecode);
+        console.log("DiamondProxy deployed at:", contractAddress);
+        addresses.stateTransition.diamondProxy = contractAddress;
+    }
+
+    function deployErc20BridgeProxy() internal {
+        bytes memory bytecode = abi.encodePacked(
+            type(TransparentUpgradeableProxy).creationCode,
+            abi.encode(addresses.bridgehub.bridgehubProxy, addresses.transparentProxyAdmin, bytes(hex""))
```

Comment:
I see that this is identical to the original script in the dev branch, but it seems wrong to create a proxy that points to another proxy cc @kelemeno